src/envs/cover.py

```diff
@@ -625,48 +622,60 @@ def simulate(self, state: State, action: Action) -> State:
                         y2) in zip(held_rect.vertices, next_held_rect.vertices)
                 ]
 
-        # Ensure neither the gripper nor the possible held block go below the
-        # y-axis.
-        if y + dy < 0 - self.collision_threshold:
-            return state.copy()
+        # Prevent the robot from going below the top of the blocks.
```

Comment:
with all these snapping hacks, are we giving up hope that option learning could work in pybullet?

Reply:
hope is diminished, but some aspects of pybullet are easier than what we had here before, e.g., there are natural limits on the robot position (kinematics and table collisions)